### ISSUE TITLE
Name generation not aware of newly generated names

### ISSUE THREAD

**Issue by OriRoth** [SpartanRefactoring/Spartanizer#1068]
The problem occur when trying to expand several code points at once:
```Java
new MyObject();
new MonaLisa();
```
==>
```Java
MyObject m = new MyObject();
MonaLisa m = new MonaLisa();
```
==>
compilation error.
#1044 
@dormaayan

**Comment by mdoron** [comment]
@OriRoth Please write an example of the code that run all the Zoomers together - How would you like the API be like?

**Comment by yossigil** [comment]
@mdoron: This is design work. Can you suggest something?

**Comment by mdoron** [comment]
@yossigil I already suggested something to @OriRoth, who was unhappy with my solution. I also discussed @dormaayan about this issue and explained my solution to him, I think he agreed with me.
But, both @dormaayan and I didn't fully understand why @OriRoth was unsatisfied with this solution.

My solution is to simply to create a new class which will implement "scope" and keep a Namespace as a private field...
